app/src/org/commcare/utils/NotificationUtil.java

```diff
@@ -0,0 +1,48 @@
+package org.commcare.utils;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+import androidx.core.app.NotificationCompat;
+
+import org.commcare.activities.DispatchActivity;
+import org.commcare.dalvik.R;
+
+/**
+ * Set of methods to post notifications to the user.
+ *
+ * @author avazirna
+ */
+public class NotificationUtil {
+    public static void showNotification(Context context, String notificationChannel, int notificationId,
```

Comment:
can use `NotificationManager.reportNotificationMessage()` ?

Reply:
As discussed, because the goal is to go back to the recording the notification action can't be to go to the `MessageActivity`. This utility class is also to be used to consolidate other places where we post notifications.

Reply:
I see, is it possible to reuse this method in `NotificationManager.reportNotificationMessage()` that case. Just wondering if there is a way to avoid code duplication.

Reply:
Yes, the plan is to refactor everywhere where we have this code to this utility class in a separate PR.